src/olympia/scanners/tests/test_actions.py

```diff
@@ -377,6 +377,29 @@ def test_runs_delay_auto_approval_indefinitely(
         assert _delay_auto_approval_indefinitely_mock.called
         _delay_auto_approval_indefinitely_mock.assert_called_with(self.version)
 
+    @mock.patch('olympia.scanners.models._delay_auto_approval_indefinitely')
+    def test_returns_for_non_extension_addons(
+        self, _delay_auto_approval_indefinitely_mock
+    ):
+        self.scanner_rule.update(action=DELAY_AUTO_APPROVAL_INDEFINITELY)
+        self.version.addon.update(type=amo.ADDON_DICT)
+
+        ScannerResult.run_action(self.version)
+
+        assert not _delay_auto_approval_indefinitely_mock.called
+
+        self.version.addon.update(type=amo.ADDON_LPAPP)
+
+        ScannerResult.run_action(self.version)
+
+        assert not _delay_auto_approval_indefinitely_mock.called
+
+        self.version.addon.update(type=amo.ADDON_STATICTHEME)
+
+        ScannerResult.run_action(self.version)
+
+        assert not _delay_auto_approval_indefinitely_mock.called
```

Comment:
It might be unnecessary, but would it be worth updating the addon type one final time to be an extension and then asserting that the mock is in fact called?